### PR TITLE
Task/update unit tests

### PR DESCRIPTION
**Pull request by mkmbayona**
Fixes and updates for unit tests

### CODE REVIEW

File: OrdermentumSDKTests/Resources/validation.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    public var totalCost: Float = 0.00
+    public var totalFreight: Float = 0.00
+    public var subtotal: Float = 0.00
+    public var totalGST: Float = 0.00
+    public var total: Float = 0.00
+    public var totalPaid: Float = 0.00
+    public var totalDue: Float = 0.00
+    public var totalFreightIncGST: Float = 0.00
+    public var surcharge: Float = 0.00
+}
```
Comment (bstillitano) [inline OrdermentumSDKTests/Resources/validation.json]: This is invalid JSON

File: OrdermentumSDKTests/Resources/Auth/changePassword.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{"success": true, message: "Change password succeeded"}
```
Comment (bstillitano) [inline OrdermentumSDKTests/Resources/Auth/changePassword.json]: Invalid JSON

File: OrdermentumSDK/Service.swift [hunk 3]
```diff
@@ -10,28 +10,34 @@ import Foundation
...
                     break
-                    
                 case .failure:
                     //This catches successful requests that return an empty body. Alamofire has a known limitation where it will fail when .validate() is added to a request with an "Accept" header.
                     if (200 ... 299).contains(response.response?.statusCode ?? 0) {
-                        completion(true, nil, nil)
+                        //For some reason the testLogin goes here with statusCode 200 and a response data
+                        //The old completion returns a success without a responseObject which fails the test
+                        //Had to add a decode here for testLogin to succeed
+                        self.doDecode(response: response, completion: completion)
```
Comment (bstillitano) [inline OrdermentumSDK/Service.swift]: Will this cause all false positives in the client?

File: OrdermentumSDK/Service.swift [hunk 4]
```diff
@@ -10,28 +10,34 @@ import Foundation
...
+                completion(true, nil, nil)
+                return
+            }
+            let responseObject = try JSONDecoder().decode(D.self, from: responseData)
+            completion(true, responseObject, nil)
+        } catch {
+            completion(true, nil, nil)
+        }
+    }
+
```
Comment (bstillitano) [inline OrdermentumSDK/Service.swift]: Not sure this is required as a separate function given that we shouldn't really be decoding inside a failure block.

File: OrdermentumSDKTests/Resources/validation.json (hunk 1 above)
Comment (mkmbayona) [reply OrdermentumSDKTests/Resources/validation.json]: Done

File: OrdermentumSDKTests/Resources/Auth/changePassword.json (hunk 2 above)
Comment (mkmbayona) [reply OrdermentumSDKTests/Resources/Auth/changePassword.json]: Done

File: OrdermentumSDK/Service.swift (hunk 3 above)
Comment (mkmbayona) [reply OrdermentumSDK/Service.swift]: Reverted

File: OrdermentumSDK/Service.swift (hunk 4 above)
Comment (mkmbayona) [reply OrdermentumSDK/Service.swift]: Reverted

**Pull request closed without merging**